# Comparative Genomics and Phylogenetic Analysis in KBase

![](../../.gitbook/assets/comparative.jpg) KBase’s comparative genomics and phylogenetic analysis tools enable researchers to understand evolutionary relationships between organisms and explore structural and functional variance across genomes. By integrating public datasets with user data and tools to search for sequence homology, explore gene orthology, and construct phylogenetic linkages between organisms, KBase provides an advanced and useful platform for comparative genomics research.

### **Phylogenetic Analysis**

* Reconstruct a phylogenetic tree based on conserved orthologous groups using [Insert Genome Into Species Tree](https://narrative.kbase.us/#appcatalog/app/l.m/insert_set_of_genomes_into_species_tree_generic)
* Examine functional content using _View Function Profile for Genomes_ and _View Function Profile Phylo_
* Compare proteomes and produce a synteny map dot plot matrix and table of gene differences with [Compare Two Proteomes](https://narrative.kbase.us/#appcatalog/app/l.m/compare_two_proteomes_generic)
* View the pangenome in a phylogenetic context using _Phylogenetic Pangenome Accumulation_

### Sequence Homology

* Search and compare translated protein sequences via [BLASTx](https://narrative.kbase.us/#catalog/apps/kb_blast/BLASTx_Search/release), [tBLASTn](https://narrative.kbase.us/#catalog/apps/kb_blast/tBLASTn_Search/release), and [tBLASTx](https://narrative.kbase.us/#catalog/apps/kb_blast/tBLASTx_Search/release)
* Search and compare nucleotide sequences via [BLASTn](https://narrative.kbase.us/#catalog/apps/kb_blast/BLASTn_Search/release)
* Search and compare protein sequences via [BLASTp](https://narrative.kbase.us/#catalog/apps/kb_blast/BLASTp_Search/release) and [psiBLAST](https://narrative.kbase.us/#catalog/apps/kb_blast/psiBLAST_msa_start_Search/release)
* Build and analyze multiple sequence alignments \(MSA\) for [nucleotide](https://narrative.kbase.us/#catalog/apps/kb_muscle/MUSCLE_nuc/release) or [protein](https://narrative.kbase.us/#catalog/apps/kb_muscle/MUSCLE_prot/release) sequences via MUSCLE

### **Pangenome Exploration** 

*  [Build Pangenome with OrthoMCL](https://narrative.kbase.us/#appcatalog/app/PangenomeOrthomcl/build_pangenome_with_orthomcl/release) – Group orthologous protein sequences and identify core and non-core genes across multiple species using
* [Compute Pangenome](https://narrative.kbase.us/#appcatalog/app/l.m/compute_pangenome) – Build a pangenome and evaluate protein family conservation with 
* _Pangenome Circle Plot_  – View a microbial pangenome as a circle plot using  
* [Compare Genomes from Pangenome](https://narrative.kbase.us/#appcatalog/app/l.m/genome_comparison_from_pangenome)  __– Compare isofunctional and homologous gene families within a pangenome with 

### Comparative and Phylogenetic Analysis Tutorials

* [Build a Gene Tree ](https://narrative.kbase.us/narrative/ws.22290.obj.1)– guides the user through the process of building a gene tree for the "Sulfate Adenylyltransferase" \(Sat\) gene from Candidatus Desulforudis audaxviator MP104C
* [Phylogenomics of Sulfate Reducing Clostridia - Tutorial - Part 1: Comparative Functional Assignment](https://narrative.kbase.us/narrative/ws.18988.obj.1) – guides the user through the process of whole-genome phylogeny, homology, and domain family functional profiling.

![](https://kbase.us/wp-content/uploads/2015/02/CompareProteomes06.png)

