# Comparative Genomics and Phylogenetic Analysis in KBase

![](../../.gitbook/assets/comparative.jpg) KBaseâ€™s comparative genomics and phylogenetic analysis tools enable researchers to understand evolutionary relationships between organisms and explore structural and functional variance across genomes. By integrating public datasets with user data and tools to search for sequence homology, explore gene orthology, and construct phylogenetic linkages between organisms, KBase provides an advanced and useful platform for comparative genomics research.

#### KBase Comparative Genomics and Phylogenetic Analysis Capabilities

**Phylogenetic Analysis**

* Reconstruct a phylogenetic tree based on conserved orthologous groups using _Insert Genome into Species Tree_
* Examine functional content using _View Function Profile for Genomes_ and _View Function Profile Phylo_
* Compare proteomes and produce a synteny map dot plot matrix and table of gene differences with _Compare Two Proteomes_
* View the pangenome in a phylogenetic context using _Phylogenetic Pangenome Accumulation_

**Sequence Homology**

* Search and compare translated protein sequences via BLASTx, tBLASTn, and tBLASTx
* Search and compare nucleotide sequences via BLASTn
* Search and compare protein sequences via BLASTp and psiBLAST
* Build and analyze multiple sequence alignments \(MSA\) for nucleotide or protein sequences via MUSCLE

**Pangenome Exploration**

* Group orthologous protein sequences and identify core and non-core genes across multiple species using _Build Pangenome with OrthoMCL_
* Build a pangenome and evaluate protein family conservation with _Compute Pangenome_
* View a microbial pangenome as a circle plot using _Pangenome Circle Plot_
* Compare isofunctional and homologous gene families within a pangenome with Compare _Genomes from Pangenome_

### Resources for comparative and phylogenetic analysis in KBase

 **App Details**

* [Insert Genome Into Species Tree](https://narrative.kbase.us/#appcatalog/app/l.m/insert_set_of_genomes_into_species_tree_generic)
* [Build Pangenome with OrthoMCL](https://narrative.kbase.us/#appcatalog/app/PangenomeOrthomcl/build_pangenome_with_orthomcl/release)
* [Compare Genomes from Pangenome](https://narrative.kbase.us/#appcatalog/app/l.m/genome_comparison_from_pangenome)
* [Compute Pangenome](https://narrative.kbase.us/#appcatalog/app/l.m/compute_pangenome)
* [Compare Two Proteomes](https://narrative.kbase.us/#appcatalog/app/l.m/compare_two_proteomes_generic)
* [Sequence Homology Search](https://narrative.kbase.us/search/#/homology/)

**Narrative Tutorials**

* [Building a gene tree](https://narrative.kbase.us/narrative/ws.22290.obj.1)
* [Whole-genome phylogeny, homology, and domain family functional profiling](https://narrative.kbase.us/narrative/ws.18988.obj.1)

![](https://kbase.us/wp-content/uploads/2015/02/CompareProteomes06.png)

