# Assembling and Annotating Microbial Genomes

![Assemble &amp; Annotate](https://kbase.us/wp-content/uploads/2016/09/genome.jpg)KBase enables _de novo_ assembly of prokaryotic NGS reads from various sequencing platforms. Assemblies can then be annotated with RAST or Prokka, enabling you to explore structural and functional features of a Genome or use it in other analyses. 

The steps for assembly and annotation include:

* Find a set of reads to assemble using the Narrative Interface Data Browser.
* Add the reads to our Narrative.
* Find and insert the Assemble Contigs from Reads app into our Narrative.
* Use this app to assemble an example set of reads into a set of contigs \(an Assembly object\).
* Find and insert the Annotate Microbial Contigs app into our Narrative.
* Use this app to annotate our Assembly.
* Examine assembly job statistics and genome annotation.

## Narrative Tutorials

* [Microbial Genome Assembly and Annotation Tutorial](https://narrative.kbase.us/narrative/ws.18188.obj.6) – guide on assembling a set of NGS short reads into contigs and annotate the assembled contigs, assigning biological functions derived from RAST 
* [Genome Analysis Tools and Features](https://narrative.kbase.us/narrative/48493) – a how-to-guide for various tools to analyze genomic sequencing data in KBase from quality control and assessment of NGS reads, de novo assembly, and genome annotation.

