# Metagenomic & Community Analysis

KBase has a multitude of tools designed to analyze [microbial communities](https://kbase.us/applist/#Microbial%20Communities) and [metagenomic reads](https://kbase.us/applist/#Comparative%20Genomics) data. They provide a way to understand the functional interactions between species in microbial communities. Annotated genomes extracted from metagenomic samples can be used for metabolic modeling, comparative phylogenomics, functional profiling, and more.

## Narrative Tutorials

* [Genome Extraction from Shotgun Metagenome Sequence Data](https://narrative.kbase.us/narrative/33233) – this is a good way to understand an important metagenomic pipeline.
* [Using Metagenomes to Discover Novel Microbial Linages in KBase](https://narrative.kbase.us/narrative/64677) –  demonstrates the application of bioinformatic tools for metagenome assembly and genome binning. This Narrative highlights the effective use of employing multiple genome binners and a bin optimization approach. __
* \_\_[The PlantSEED Resource in KBase](https://narrative.kbase.us/narrative/39144) – shows the streamlined process of annotating plant genome sequences, automatically constructing metabolic models based on genome annotations, and using models to test annotations.

